ee/src/main/java/org/jboss/as/ee/component/ViewDescription.java

```diff
@@ -179,12 +179,25 @@ public Deque<ViewConfigurator> getConfigurators() {
      * Create the injection source
      *
      * @param serviceName     The view service name
-     * @param viewClassLoader
+     * @param viewClassLoader The view class loader
+     * @param appclient       appclient environment
+     * @deprecated use {@link ViewDescription#createInjectionSource(ServiceName, Supplier, boolean)} instead. This method will be removed in the future.
      */
     protected InjectionSource createInjectionSource(final ServiceName serviceName, Value<ClassLoader> viewClassLoader, boolean appclient) {
         return new ViewBindingInjectionSource(serviceName);
     }
 
+    /**
+     * Create the injection source
+     *
+     * @param serviceName     The view service name
+     * @param viewClassLoader The view class loader
+     * @param appclient       appclient environment
+     */
+    protected InjectionSource createInjectionSource(final ServiceName serviceName, Supplier<ClassLoader> viewClassLoader, boolean appclient) {
+        return new ViewBindingInjectionSource(serviceName);
+    }
```

Comment:
this method only uses the first param and ignoring the other 2 params. Can we just remove the 2 params?

Reply:
We cannot. It is template method overwritten by [EJBViewDescription](https://github.com/wildfly/wildfly/blob/main/ejb3/src/main/java/org/jboss/as/ejb3/component/EJBViewDescription.java#L105).